### PR TITLE
Replace fitting code with standard scipy.optimize fitters

### PR DESCRIPTION
**Pull request by joshmutus**
Removed a lot of custom fitting code in favor of features built into `scipy.optimize` this doesn't totally replace fuctionality of existing code, but I think this is a simple place from which to start re-adding old functionality.

This also serves as a very clear minimum working example for new participants to the repo

### CODE REVIEW

File: resfit/fit_resonator/resonator.py [hunk 1]
```diff
@@ -301,15 +301,15 @@ def __init__(self,
...
+            self.func = fit_functions.cavity_DCM_REFLECTION
         elif method.name == 'PHI':
-            self.func = fit_functions.Cavity_DCM
+            self.func = fit_functions.cavity_DCM
         elif method.name == 'INV':
-            self.func = fit_functions.Cavity_inverse
+            self.func = fit_functions.cavity_inverse
         elif method.name == 'CPZM':
-            self.func = fit_functions.Cavity_CPZM
+            self.func = fit_functions.cavity_CPZM
```
Comment (kunalq) [inline resfit/fit_resonator/resonator.py]: Up to you, but you can also consider moving this into the enum:
```python
class FittingMethod(enum.Enum):
    DCM = 1
    DCM_REFLECTION = 2
    PHI = 3
    INV = 4
    CPZM = 5

    def get_function(self):
        if self == FittingMethod.DCM:
            return fit_functions.cavity_DCM
        elif self == FittingMethod.DCM_REFLECTION:
            return fit_functions.cavity_DCM_REFLECTION
        elif self == FittingMethod.PHI:
            return fit_functions.cavity_DCM
        elif self == FittingMethod.INV:
            return fit_functions.cavity_inverse
        elif self == FittingMethod.CPZM:
            return fit_functions.cavity_CPZM

        raise ValueError('Unknown fitting method: {}'.format(self))


class FitMethod(object):
    def __init__(self, ...):
        self.func = FittingMethod(method).get_function()
```

File: resfit/fit_resonator/fit_functions.py [hunk 2]
```diff
@@ -27,36 +26,38 @@ def from_params(cls, Qi,  Qc, f_res, phi, Q=None):
...
+def one_cavity_peak_abs(x, Q, Qc, w1):
     #Ideal resonator fit function
     return np.abs(Q/Qc/(1 + 1j*(x-w1)/w1*2*Q))
 
-def One_Cavity_peak_abs_REFLECTION(x, Q, Qc, w1):
+def one_cavity_peak_abs_REFLECTION(x, Q, Qc, w1):
     #Ideal resonator fit function
-    return np.abs(2*Q/Qc/(1 + 1j*(x-w1)/w1*2*Q))
\ No newline at end of file
+    return np.abs(2*Q/Qc/(1 + 1j*(x-w1)/w1*2*Q))
```
Comment (kunalq) [inline resfit/fit_resonator/fit_functions.py]: Doesn't look like either of the `one_cavity_` methods are used -- maybe add them to the enum?

File: resfit/fit_resonator/resonator.py (hunk 1 above)
Comment (joshmutus) [reply resfit/fit_resonator/resonator.py]: For some reason the enum won't evaluate itself correctly. If I create
```python
fit = FittingMethod.DCM
fit.get_function()
>>>Unknown fitting method: FittingMethod.DCM
```

And if I just put `self == FittingMethod.DCM` evaluates to `false`

I'm going to punt this for a future PR

File: resfit/fit_resonator/fit_functions.py [hunk 3]
```diff
@@ -27,36 +26,38 @@ def from_params(cls, Qi,  Qc, f_res, phi, Q=None):
...
 
-def Cavity_DCM_REFLECTION(x, Q, Qc, w1, phi):
+def cavity_DCM_REFLECTION(x, Q, Qc, w1, phi):
     #DCM REFLECTION fit function
     return np.array(1-2*Q/Qc*np.exp(1j*phi)/(1 + 1j*(x-w1)/w1*2*Q))
 
-def Cavity_inverse(x, Qi, Qc, w1, phi):
+def cavity_inverse(x, Qi, Qc, w1, phi):
     #Inverse fit function
     return np.array(\
```
[the remark this answers is not in the document]
Comment (joshmutus) [reply resfit/fit_resonator/fit_functions.py]: thank you :)

File: resfit/fit_resonator/fit_functions.py [hunk 4]
```diff
@@ -27,36 +26,38 @@ def from_params(cls, Qi,  Qc, f_res, phi, Q=None):
...
-    INV = 4
-    CPZM = 5
-
-def Cavity_DCM(x, Q, Qc, w1, phi):
+def cavity_DCM(x, Q, Qc, w1, phi):
     #DCM fit function
     return np.array(1-Q/Qc*np.exp(1j*phi)/(1 + 1j*(x-w1)/w1*2*Q))
 
-def Cavity_DCM_REFLECTION(x, Q, Qc, w1, phi):
+def cavity_DCM_REFLECTION(x, Q, Qc, w1, phi):
```
[the remark this answers is not in the document]
Comment (joshmutus) [reply resfit/fit_resonator/fit_functions.py]: I'll pick that up later. What's the rule for acronyms in lower_case naming conventions?

**Pull request merged**